yb-voyager/cmd/exportData.go

```diff
@@ -241,6 +241,23 @@ func exportData() bool {
 
 	// Check if source DB has required permissions for export data
 	if source.RunGuardrailsChecks {
+		// If source is PostgreSQL or YB, check if the number of existing replicaton slots is less than the max allowed
+		if (source.DBType == POSTGRESQL && changeStreamingIsEnabled(exportType)) ||
+			(source.DBType == YUGABYTEDB && !bool(useYBgRPCConnector)) {
+			// The queries used to check replication slots on YB don't throw an error even if logical replication is not supported
+			// in that YB version. They are inbuilt PG queries and will return the default values i.e. max allowed slots = 10 and
+			// current slots = 0
+			// Hence it is safe to check this in YB irrespective of the version.
+			result, err := source.DB().CheckReplicationSlots()
```

Comment:
How about `CheckIfReplicationSlotsAreAvailable` which returns `(bool, int, int, error)` which is essentially `(isAvailable, usedCount, maxCount, error)` ?
It doesn't seem very clean for these functions to return a "string". exportData layer can frame the error message. 

Same applies for the other functions, `GetMissingExportSchemaPermissions`, for example, ideally, the db functiouns should just return details about the permission type, and the list of objects missing that permission . (so, perhaps a map[permissionType][]objects). The exportData layer should frame the error messages. (can take this up in a separate later PR)